### PR TITLE
Fix: _syncHandle sometimes causing NPEs in Kotlin / Wasm code

### PR DESCRIPTION
**Pull request by alxrsngrtn**
When onHandleSync is updated and there is no model, it causes a NPE error in Kotlin / Wasm code. 
This manifests as a runtime error in the browser console and is masked as an `abort` error. 

This PR attempts to fix the issue by taking a more defensive approach. However, I am unsure of the effects it will have on C++ / Wasm code.

### CODE REVIEW

File: src/runtime/wasm.ts [hunk 1]
```diff
@@ -621,7 +621,7 @@ export class WasmParticle extends Particle {
   async onHandleSync(handle: Handle, model) {
     const wasmHandle = this.handleMap.getL(handle);
     if (!model) {
-      this.exports._syncHandle(this.innerParticle, wasmHandle, 0);
+      this.exports._syncHandle(this.innerParticle, wasmHandle, null);
```
Comment (mykmartin) [inline src/runtime/wasm.ts]: This should be left as 0. Wasm doesn't have a concept of "null" at the type level, so we should explicitly set 0 to be interpreted as a null pointer internally.

**Comment by alxrsngrtn** [comment]
The Wasm particles still should be failing the manual tests. I believe that this is due to a collections encoding error. For instance, the following seems malformed: 
`Handle is 'inputData' syncing '1:111:85:!846534373238833:https://$particles/Tutorial/Kotlin/5_Collections/Collections.arcs::0|name:T4:Jill|age:N70:|'`

**Comment by mykmartin** [comment]
I presume this is using a Kotlin particle with `particles/Tutorial/Javascript/5_Collections/Collections.arcs`?

Assuming that, the encoding looks correct. The collection format is `<num-entities>:<length>:<encoded><length>:<encoded>...`

`1:111:85:!846534373238833:https://$particles/Tutorial/Kotlin/5_Collections/Collections.arcs::0|name:T4:Jill|age:N70:|`

So num-entities = 1 and length = 111, which matches the length of the remaining string. Then there should be one `<encoded>` entry (because num-entities = 1). That will be a singleton encoding, which is `<id-length>:<id>|<field>:<value>|<field>:<value>|...|`.
```
id-length = 85
id = !846534373238833:https://$particles/Tutorial/Kotlin/5_Collections/Collections.arcs::0
field = 'name'
value = 'T4:Jill' -> Text, length 4, 'Jill'
field = 'age'
value = 'N70' -> Number, 70
```

**Comment by alxrsngrtn** [comment]
@mykmartin Found the issue -- the kotlinic syntax iterated one too many times.

**Comment by alxrsngrtn** [comment]
Cool, will merge after build passes.

**Pull request merged**